src/serialization/QuerySerializer.ts

```diff
@@ -0,0 +1,10 @@
+import QueryRepresentation from '@/sparql/QueryRepresentation';
+
+export default class QuerySerializer {
+	public serialize( query: QueryRepresentation ): object {
```

Comment:
I'm sorry, but you can't start with a `QueryRepresentation`. The `query` getter (rightly) assumes an already valid Query Builder, but we want to be able to generate this link even with just a partially filled query builder.
So the input should be the root state itself. That being said, it is probably enough to serialize the same information that will end up in the `QueryRepresentation`, i.e. no need to serialize errors or labels or the `conditionId`.

Reply:
Ok! Makes sense too.